File: Application/lib/model/order_pending.dart
import 'package:shopping_app/model/sale.dart';

class OrderPending {
  int? orderPendingId;
  int? customerId;
  String? employeeId;
  String? customerName;
  String? employeeName;
  String? odertDate;
  String? status;
  List<ListProductOrederPending>? listProductOrederPending;

  OrderPending({
    this.orderPendingId,
    this.customerId,
    this.employeeId,
    this.customerName,
    this.employeeName,
    this.odertDate,
    this.status,
    this.listProductOrederPending,
  });

  OrderPending.fromJson(Map<String, dynamic> json) {
    orderPendingId = json['orderPendingId'];
    customerId = json['customerId'];
    employeeId = json['employeeId'];
    customerName = json['customerName'];
    employeeName = json['employeeName'];
    odertDate = json['odertDate'];
    status = json['status'];
    if (json['listProductOrederPending'] != null) {
      listProductOrederPending = <ListProductOrederPending>[];
      json['listProductOrederPending'].forEach((v) {
        listProductOrederPending!.add(ListProductOrederPending.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['orderPendingId'] = orderPendingId;
    data['customerId'] = customerId;
    data['employeeId'] = employeeId;
    data['customerName'] = customerName;
    data['employeeName'] = employeeName;
    data['odertDate'] = odertDate;
    data['status'] = status;
    if (listProductOrederPending != null) {
      data['listProductOrederPending'] =
          listProductOrederPending!.map((v) => v.toJson()).toList();
    }
    return data;
  }
}

class ListProductOrederPending {
  int? productOrderPendingId;
  int? productId;
  String? productName;
  int? amount;
  String? colorName;
  String? brand;
  String? series;
  double? price;
  String? categoryName;
  String? mainImg;
  Sale? sale;

  ListProductOrederPending(
      {this.productOrderPendingId,
      this.productId,
      this.productName,
      this.amount,
      this.colorName,
      this.brand,
      this.series,
      this.price,
      this.categoryName,
      this.mainImg,
      this.sale});

  ListProductOrederPending.fromJson(Map<String, dynamic> json) {
    productOrderPendingId = json['productOrderPendingId'];
    productId = json['productId'];
    productName = json['productName'];
    amount = json['amount'];
    colorName = json['colorName'];
    brand = json['brand'];
    series = json['series'];
    price = json['price'];
    categoryName = json['categoryName'];
    mainImg = json['mainImg'];
    sale = json['sale'] != null ? Sale.fromJson(json['sale']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['productOrderPendingId'] = productOrderPendingId;
    data['productId'] = productId;
    data['productName'] = productName;
    data['amount'] = amount;
    data['colorName'] = colorName;
    data['brand'] = brand;
    data['series'] = series;
    data['price'] = price;
    data['categoryName'] = categoryName;
    data['mainImg'] = mainImg;
    if (sale != null) {
      data['sale'] = sale!.toJson();
    }
    return data;
  }
}

File: Admin/src/model/OrderPending.ts
class ProductOrderPending {
    public productOrderPendingId: number;
    public productName: string;
    public amount: number;

    public constructor(data: {
        productOrderPendingId: number;
        productName: string;
        amount: number;
    }) {
        this.productOrderPendingId = data.productOrderPendingId;
        this.productName = data.productName;
        this.amount = data.amount;
    }
}

class OrderPending {
    public orderPendingId: number;
    public customerName: string;
    public employeeName: string | null;
    public orderDate: Date;
    public status: string;
    public listProductOrderPending: ProductOrderPending[];

    public constructor(data: {
        orderPendingId: number;
        customerName: string;
        employeeName: string | null;
        orderDate: string;
        status: string;
        listProductOrederPending: {
            productOrderPendingId: number;
            productName: string;
            amount: number;
        }[];
    }) {
        this.orderPendingId = data.orderPendingId;
        this.customerName = data.customerName;
        this.employeeName = data.employeeName;
        this.orderDate = new Date(data.orderDate);
        this.status = data.status;
        this.listProductOrderPending = data.listProductOrederPending.map(
            (product) => new ProductOrderPending(product),
        );
    }
}

export default OrderPending;
